### ISSUE TITLE
Web eval fails on non-nullable variables?

### ISSUE THREAD

**Issue by vsmenon** [dart-lang/webdev#1178]
I'm running a Flutter Web app on master, ported to null safety.  If I step past an assignment to a non-null typed variable and try to evaluate it, I get an error.

E.g., I'm stepping past:

```dart
var choices = buildChoices(); // Returns a non-null List
```

On the variables window, I can see `choices` and drill into its contents.  If I enter:
```> choices```
in the REPL, I get:

```
CompilationError: Non-nullable variable 'choices' must be assigned before it can be used.
choices
^^^^^^^
```

If I hover on `choices` (in VSCode) if doesn't do anything (I assume silently swallowing the exception).  This appears to be an error triggered by the CFE.  FWIW, I don't see any error (i.e., it works as expected) on Flutter mobile.

This is on Flutter / Dart version:
```
Flutter 1.24.0-8.0.pre.117 • channel master • https://github.com/flutter/flutter.git
Framework • revision 8cb2665118 (14 hours ago) • 2020-11-06 16:02:19 +0800
Engine • revision 0693ee04d1
Tools • Dart 2.12.0 (build 2.12.0-21.0.dev)
```

I'm using VSCode - Dart/Flutter plugin 3.16.0.

**Comment by vsmenon** [comment]
FYI - @annagrin @grouma - this might start coming up more as we roll out null safety.

**Comment by vsmenon** [comment]
fyi - @mariamhas @johnpryan - we'll need to start switching the FW CUJ tests to null safe versions.

**Comment by annagrin** [comment]
@johnniwinther this is the issue you mentioned as a problem with flow analysis when compiling the synthetic function created in IncrementalCompiler.compileExpression.

**Comment by annagrin** [comment]
Created the issue in the SDK. Will keep this one open to confirm the fix in dwds/flutter after the SDK issue is resolved.

**Comment by annagrin** [comment]
- confirmed as fixed in webdev.
- will be fixed in flutter as soon dwds version 7.1.1 is published and flutter picks it up.

**Comment by annagrin** [comment]
Verified - fixed in flutter master:

Flutter 1.24.0-8.0.pre.337 • channel master • git@github.com:flutter/flutter.git
Framework • revision c5f8edd9ee (51 minutes ago) • 2020-11-20 15:20:09 -0800
Engine • revision 23a8e027db
Tools • Dart 2.12.0 (build 2.12.0-62.0.dev)
